src/components/Select/Select.tsx

```diff
@@ -119,6 +119,15 @@ const Select: React.FunctionComponent<SelectProps> = ({ options, onChange }) =>
     }
   }
 
+  useEffect(() => {
+    const updatedSelection = options.find(
+      (option) => selectedOption.value === option.value && selectedOption.label !== option.label,
+    )
+    if (updatedSelection) {
+      setSelectedOption(updatedSelection)
+    }
+  }, [options, selectedOption])
+
```

Comment:
I understand that this fixes the problem but it is not an optimal solution. This is essentially a hook specifically for the translation function.

I am not sure if this would work but I think it would be better for `selectedOption` to be the index of the options instead. That way we are using the latest item in options.

Can you test that theory?

Reply:
Thanks, I applied and it works. 👍